vault/mount.go

```diff
@@ -350,19 +384,80 @@ func (c *Core) remount(src, dst string) error {
 	return nil
 }
 
+// tuneMount is used to set config on a mount point
+func (c *Core) tuneMount(path string, config MountConfig) error {
+	// Ensure we end the path in a slash
+	if !strings.HasSuffix(path, "/") {
+		path += "/"
+	}
+
+	// Prevent protected paths from being changed
+	for _, p := range protectedMounts {
+		if strings.HasPrefix(path, p) {
+			return fmt.Errorf("cannot tune '%s'", path)
+		}
+	}
+
+	// Verify exact match of the route
+	match := c.router.MatchingMount(path)
+	if match == "" || path != match {
+		return fmt.Errorf("no matching mount at '%s'", path)
+	}
+
+	// Find and modify mount
+	for _, ent := range c.mounts.Entries {
+		if ent.Path == path {
+			if config.MaxLeaseTTL != nil {
+				if *config.MaxLeaseTTL == 0 {
+					*ent.Config.MaxLeaseTTL = 0
+				} else {
+					ent.Config.MaxLeaseTTL = config.MaxLeaseTTL
+				}
+			}
+			if config.DefaultLeaseTTL != nil {
+				if *ent.Config.MaxLeaseTTL == 0 {
+					if *config.DefaultLeaseTTL > c.maxLeaseTTL {
```

Comment:
Is it an issue for default of a backend (e.g. PKI) to be greater than max of global? I'm thinking global max 7 days, PKI default maybe 30days or something